ui/hooks/usePolling.ts

```diff
@@ -0,0 +1,52 @@
+import { useEffect, useRef } from 'react';
+
+type UsePollingOptions = {
+  callback?: (pollingToken: string) => (pollingToken: string) => void;
+  startPollingByNetworkClientId: (
+    networkClientId: string,
+    options: any,
+  ) => Promise<string>;
+  stopPollingByPollingToken: (pollingToken: string) => void;
+  networkClientId: string;
+  options?: any;
+};
+
+const usePolling = (
+  usePollingOptions: UsePollingOptions,
+  dependencies: unknown[] = [],
+) => {
+  const pollTokenRef = useRef<null | string>(null);
+  const cleanupRef = useRef<null | ((pollingToken: string) => void)>(null);
+  let isMounted = false;
```

Comment:
since these aren't referenced outside of the useEffect, does it make sense to bring them into the useEffect scope and use a regular variable instead of a useRef?